web/beacon-app/src/components/ui/PasswordField/usePasswordField.tsx

```diff
@@ -15,7 +15,7 @@ export const usePasswordField = (props: PasswordFieldProps): PasswordFieldProps
     ...props,
     type: showPassword ? 'text' : 'password',
     rightIcon: (
-      <button onClick={handleShowPassword} data-testid="button">
+      <button onClick={handleShowPassword} data-testid="button" type="button">
```

Comment:
When the eye icon is clicked without the type="button", it behaves as a submit button.